### ISSUE TITLE
Analyzers report weird before protocol has been confirmed

### ISSUE THREAD

**Issue by bbannier** [zeek/spicy-plugin#101]
Currently we often emit weird log entries if we see parse errors during explorative processing in `PacketAnalyzer::AnalyzePacket` and `ProtocolAnalyzer::[Process|Finish]`] (we also report weirds from `FileAnalyer` which might be different). If such reporting happens during DPD before confirmation parse errors are to be expected. Reporting them to the weird log on the other hand makes it look like there is a problem which needs to be investigated; we also make it hard to distinguish genuine parse errors (after DPD confirmation) from ones during DPD.

We should look how we can reduce weirds during DPD. We could e.g., track whether an analyzer has confirmed a protocol and suppress or redirect during DPD.

**Comment by rsmmr** [comment]
Yeah, I also think generally reporting this to `weird` isn't right as it doesn't match the level of information that the old-style analyzers record there. For them it's an explicit decision to record a `weird`, along with a short, crisp identifier describing semantically what's going on. Spicy, on the other hand, simply reports all parse errors along with low-level information that's not very helpful to users.

The main reason for the current state is actually simply that Zeek has no other way to report parse errors. Maybe we should create that. How about this:

- We record parse errors in a separate log file, say `spicy.log` or `analyzers.log`. We'd probably want to make creation of that optional too in case people aren't interested.

- We only record errors happening after confirmation.

- We provide a function `zeek::weird()` that analyzer code can use to explicit create a `weird.log` entry, just like the old analyzers do it.

**Comment by 0xxon** [comment]
With binpac parse errors after confirmation end up in dpd.log. I kind of always liked that - and if the format fits, this would be great for after-confirmation-parse errors. It also is where people will look.

For errors during exploration I agree that a separate logfile would be neat. And I also agree that providing a weird function sounds like a good idea.

**Comment by rsmmr** [comment]
Ah, didn't think about DPD log, good point. Iirc, we currently log into that when DPD receives a protocol violation, which indeed would fit.

**Comment by rsmmr** [comment]
So I think this works for protocol analyzers, but I don't know what to do for packet and file analyzers where Zeek doesn't have a reporting/suppression mechanism in place (also see https://github.com/zeek/zeek/issues/2031). I'm tempted to silently ignore them for these rather than generally log.
